client/state/sites/selectors.js

```diff
@@ -55,10 +55,13 @@ export const getRawSite = ( state, siteId ) => {
  */
 export const getSite = createSelector(
 	( state, siteId ) => {
-		const site = getRawSite( state, siteId );
+		const site = getRawSite( state, siteId ) ||
+			// Support for non-ID site retrieval
+			// Replaces SitesList#getSite
+			getSiteBySlug( state, siteId );
```

Comment:
There would be a circular dependency between `getRawSite` and `getSiteBySlug` via `getSiteSlug`.